src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -6251,5 +6251,341 @@ .maxstack  2
+            binaryOperator("bool", "<", "nuint", uintMaxValue, "nuint", uintMaxValue, "False");
+
+            binaryOperator("bool", "<=", "nint", intMinValue, "nint", intMinValue, "True");
+            binaryOperator("bool", "<=", "nint", intMaxValue, "nint", intMinValue, "False");
+            binaryOperator("bool", "<=", "nint", intMaxValue, "nint", intMaxValue, "True");
+            binaryOperator("bool", "<=", "nuint", "0", "nuint", "0", "True");
+            binaryOperator("bool", "<=", "nuint", uintMaxValue, "nuint", "0", "False");
+            binaryOperator("bool", "<=", "nuint", uintMaxValue, "nuint", uintMaxValue, "True");
+
+            binaryOperator("bool", ">", "nint", intMinValue, "nint", intMinValue, "False");
+            binaryOperator("bool", ">", "nint", intMaxValue, "nint", intMinValue, "True");
+            binaryOperator("bool", ">", "nint", intMaxValue, "nint", intMaxValue, "False");
+            binaryOperator("bool", ">", "nuint", "0", "nuint", "0", "False");
+            binaryOperator("bool", ">", "nuint", uintMaxValue, "nuint", "0", "True");
+            binaryOperator("bool", ">", "nuint", uintMaxValue, "nuint", uintMaxValue, "False");
+
+            binaryOperator("bool", ">=", "nint", intMinValue, "nint", intMinValue, "True");
+            binaryOperator("bool", ">=", "nint", intMinValue, "nint", intMaxValue, "False");
+            binaryOperator("bool", ">=", "nint", intMaxValue, "nint", intMaxValue, "True");
+            binaryOperator("bool", ">=", "nuint", "0", "nuint", "0", "True");
+            binaryOperator("bool", ">=", "nuint", "0", "nuint", uintMaxValue, "False");
+            binaryOperator("bool", ">=", "nuint", uintMaxValue, "nuint", uintMaxValue, "True");
+
+            binaryOperator("bool", "==", "nint", intMinValue, "nint", intMinValue, "True");
+            binaryOperator("bool", "==", "nint", intMinValue, "nint", intMaxValue, "False");
+            binaryOperator("bool", "==", "nint", intMaxValue, "nint", intMaxValue, "True");
+            binaryOperator("bool", "==", "nuint", "0", "nuint", "0", "True");
+            binaryOperator("bool", "==", "nuint", "0", "nuint", uintMaxValue, "False");
+            binaryOperator("bool", "==", "nuint", uintMaxValue, "nuint", uintMaxValue, "True");
+
+            binaryOperator("bool", "!=", "nint", intMinValue, "nint", intMinValue, "False");
+            binaryOperator("bool", "!=", "nint", intMinValue, "nint", intMaxValue, "True");
+            binaryOperator("bool", "!=", "nint", intMaxValue, "nint", intMaxValue, "False");
+            binaryOperator("bool", "!=", "nuint", "0", "nuint", "0", "False");
+            binaryOperator("bool", "!=", "nuint", "0", "nuint", uintMaxValue, "True");
+            binaryOperator("bool", "!=", "nuint", uintMaxValue, "nuint", uintMaxValue, "False");
+
+            binaryOperator("nint", "<<", "nint", intMinValue, "int", "0", intMinValue);
```

Comment:
>"<<" [](start = 35, length = 4)

It feels like for some values result should be platform dependent.

Reply:
Thanks. I'll address in a subsequent PR.

---
In reply to: [389955804](https://github.com/dotnet/roslyn/pull/42143#discussion_r389955804) [](ancestors = 389955804)